modules/Users/User.php

```diff
@@ -1513,6 +1513,23 @@ public function getEmailLink2(
         $class = ''
     ) {
         $emailLink = '';
+
+        require_once('modules/Emails/EmailUI.php');
+        $emailUI = new EmailUI();
+        for ($i = 0; $i < count($focus->emailAddress->addresses); $i++) {
+            $emailField = 'email' . (string) ($i + 1);
+            $optOut = (bool)$focus->emailAddress->addresses[$i]['opt_out'];
```

Comment:
I don't see the guarantee for existance of `'opt_out'` and `'email_address'` indexes. Can you please check if array indexes are exists before using it to evolve any additional and possible php error warning or notice in apache log. Use suitecrm log instead.

Reply:
I just changed the indent on this code (the gihub diff is a bit confusing imo), and I'd like to avoid unrelated changes in this PR. Should I look into this in a separate PR?

Reply:
> What do you mean "avoid unrelated changes"?

I mean that I tried to not change anything for the "sugar" client code path and I have not changed anything in this loop except removing the "if" surrounding it.

But I'll look into this.

Reply:
Yeah, please try to keep the code behaviour as it is, your changes should be backward compatible as we don't want to brake anything else, even if caller function in someones custom folder.

Reply:
Could this function called from any custom modules?

Reply:
Which function?

Maybe looking at the diff of the specific commit makes it clearer (the PR diff is confusing): https://github.com/salesagility/SuiteCRM/pull/6486/commits/001c85061d4f135cb9d9ab441b7b900bf42dc112